scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -441,46 +433,50 @@ def unpacker(res):
+            ea_str = str(ea)
+
+        try:
+            res1db = unpacker(hypotest(*data_raveled, *args,
+                                       nan_policy=nan_policy, **kwds))
+        except (RuntimeWarning, UserWarning, ValueError, ZeroDivisionError) as eb:
+            eb_str = str(eb)
+
+        try:
+            res1dc = unpacker(hypotest(*data, *args, axis=None,
+                                       nan_policy=nan_policy, **kwds))
+        except (RuntimeWarning, UserWarning, ValueError, ZeroDivisionError) as ec:
+            ec_str = str(ec)
+
+        if ea_str or eb_str or ec_str:
+            too_small_message = "See documentation for sample size requirements."
+            messages_same = ea_str == eb_str == ec_str
+            messages_ok = ((too_small_message in eb_str)
+                           and (too_small_message in ec_str))
+            assert messages_same or messages_ok
 
-            try:
-                res1db = unpacker(hypotest(*data_raveled, *args,
-                                           nan_policy=nan_policy, **kwds))
-            except (RuntimeWarning, ValueError, ZeroDivisionError) as eb:
-                eb_str = str(eb)
-
-            try:
-                res1dc = unpacker(hypotest(*data, *args, axis=None,
-                                           nan_policy=nan_policy, **kwds))
-            except (RuntimeWarning, ValueError, ZeroDivisionError) as ec:
-                ec_str = str(ec)
-
-            if ea_str or eb_str or ec_str:
-                assert any([str(ea_str).startswith(message)
-                            for message in too_small_messages])
-                assert ea_str == eb_str == ec_str
-            else:
-                assert_equal(res1db, res1da)
-                assert_equal(res1dc, res1da)
-                for item in list(res1da) + list(res1db) + list(res1dc):
-                    # Most functions naturally return NumPy numbers, which
-                    # are drop-in replacements for the Python versions but with
-                    # desirable attributes. Make sure this is consistent.
-                    assert np.issubdtype(item.dtype, np.number)
+        else:
+            assert_equal(res1db, res1da)
+            assert_equal(res1dc, res1da)
+            for item in list(res1da) + list(res1db) + list(res1dc):
+                # Most functions naturally return NumPy numbers, which
+                # are drop-in replacements for the Python versions but with
+                # desirable attributes. Make sure this is consistent.
+                assert np.issubdtype(item.dtype, np.number)
 
 # Test keepdims for:
 #     - single-output and multi-output functions (gmean and mannwhitneyu)
 #     - Axis negative, positive, None, and tuple
 #     - 1D with no NaNs
 #     - 1D with NaN propagation
 #     - Zero-sized output
+@pytest.mark.filterwarnings('ignore:All axis-slices of one...')
+@pytest.mark.filterwarnings('ignore:After omitting NaNs...')
```

Comment:
These additions go in the wrong direction; or are these just cases we haven't investigated/fixed yet?

Reply:
I think these are fine because we emit these warnings intentionally, and looking for them is not the point of _this_ test, so it seems OK to ignore them.

Reply:
Ok, if the warnings are intentional, then that's fine for me. I just want to avoid filtering out the _unintentional_ ones in the tests, rather doing it properly in the implementation

Reply:
I see. Yeah, these are the messages of the intentional warnings. Maybe I will import the variables that hold them to make it more clear.